crates/noirc_evaluator/src/ssa_refactor/ir/instruction.rs

```diff
@@ -499,7 +499,136 @@ fn simplify_call(func: ValueId, arguments: &[ValueId], dfg: &mut DataFlowGraph)
                 None
             }
         }
-        Intrinsic::BlackBox(_) | Intrinsic::Println | Intrinsic::Sort => None,
+        Intrinsic::BlackBox(bb_func) => match bb_func {
+            BlackBoxFunc::AND => {
+                if let (Some((lhs, Type::Numeric(NumericType::Unsigned { bit_size }))), Some(rhs)) = (
+                    dfg.get_numeric_constant_with_type(arguments[0]),
+                    dfg.get_numeric_constant(arguments[1]),
```

Comment:
TODO: add comment explaining that we've already checked that these types are the same. (or add a quick assert.)